sphinx_rtd_theme/layout.html

```diff
@@ -51,12 +51,29 @@
 
   {#- CANONICAL URL (deprecated) #}
   {%- if theme_canonical_url and not pageurl %}
-    <link rel="canonical" href="{{ theme_canonical_url }}{{ pagename }}.html"/>
+    <link rel="canonical" href="{{ theme_canonical_url }}{{ pagename }}{% if builder == 'dirhtml' %}/{% else %}.html{% endif %}"/>
   {%- endif -%}
 
-  {#- CANONICAL URL #}
-  {%- if pageurl %}
-    <link rel="canonical" href="{{ pageurl|e }}" />
+  {#-
+    CANONICAL URL
+    NB! pageurl is currently a non-documented template context variable!
+    pageurl implementation: https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/html.html
```

Comment:
I'd suggest a link to the GitHub source code at the correct line where this `pageurl` is injected in the context.

Reply:
:+1: